integrations/webchat/agent-assist/agent-verify/client/javascript/connectors/genesys-transcription-connector.js

```diff
@@ -0,0 +1,129 @@
+        let transcripts = data.eventBody.transcripts;
+        transcripts.forEach(t => {
+            if (t?.alternatives?.length > 0) {
+                let text = t.alternatives[0].transcript;
+
+                if (t.channel === 'INTERNAL') {
+                    window.watsonAssistantChat.agentAssist.onAgentSaid(text);
+                } else {
+                    window.watsonAssistantChat.agentAssist.onUserSaid(text);
+                }
+            }
+        });
+    }
+};
+
+function addCallSubscription(topic) {
+    let body = [{ 'id': topic }];
+
+    return notificationsApi.postNotificationsChannelSubscriptions(
+        callChannel.id, body)
+        .then((_data) => {
+            console.log(`postNotificationsChannelSubscriptions complete`);
+        });
+}
+
+/**
+ * Create the call channel. If called multiple times,
+ * the last one will be the active one.
+ */
+function createCallChannel() {
+    return notificationsApi.postNotificationsChannels()
+        .then(data => {
+            console.log('---- Created Notifications Channel ----');
+            console.log(data);
+
+            callChannel = data;
+            let ws = new WebSocket(callChannel.connectUri);
+            ws.onmessage = (event) => {
+                let dataObj = JSON.parse(event.data);
+                onTranscriptionMessage(dataObj);
+            };
+        });
+}
+
+/**
+ * Set-up the channel for call conversations
+ */
+function setupCallChannel(conversationID) {
+    return createCallChannel()
+        .then(data => {
+            // Subscribe to incoming chat conversations
+            return addCallSubscription(`v2.conversations.${conversationID}.transcription`);
+        });
+}
+
+
+function getRedirectURL() {
+    let redirectUrl = new URL(document.location.href);
+    redirectUrl.searchParams.delete('conversationID');
+    return redirectUrl.href;
+}
+    
+function initConnector() {
+    console.log('initConnector: TOP');
+
+    const urlSearchParams = new URLSearchParams(window.location.search);
+    const params = Object.fromEntries(urlSearchParams.entries());
+
+    // Get the parameters needed to work with the Genesys APIs from query parameters
+    clientId =  params.clientID || null;
+    environment =  params.environment || null;
+    agentAssistConversationID = params.conversationID || null;
+
+    // Obtain a reference to the platformClient object
+    platformClient = require('platformClient');
+    client = platformClient.ApiClient.instance;
+
+    // API instances
+    notificationsApi = new platformClient.NotificationsApi();
+    
+    client.setEnvironment(environment);
+    client.setPersistSettings(true, 'test_app');
+
+    client.loginImplicitGrant(clientId,
+        getRedirectURL(),
+        { state: agentAssistConversationID })
+        .then(data => {
+            console.log(data);
+            // Create the channel for call notifications
+            return setupCallChannel(agentAssistConversationID);
```

Comment:
Hrmmmm, are you sure this works? The value of `agentAssistConversationID` is coming from the URL parameter but once a redirect occurs, this URL parameter no longer exists. You're supposed to get this value from the state value that was passed to `loginImplicitGrant` and that is later returned to you as `data.state`. I suspect this may only be working for you sometimes because you have `setPersistSettings` which bypasses the login process if you're already logged in and thus avoids the redirect. If you comment out `setPersistSettings`, I bet this code doesn't work.

I'm going to try fixing this in my commit but you'll need to verify it still works.

Reply:
We verified that this is indeed working.